### ISSUE TITLE
Null object exception on listening to list changes

### ISSUE THREAD

**Issue by grendo** [schulz3000/deepstreamNet#28]
Hi,
  with the following test I am getting a null reference on e.OldItems in the JsonNetChangeListener.  It occurs on the second call into the CollectionChanged event.

```
        void Collection_CollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
        {
            if (e.Action == NotifyCollectionChangedAction.Replace || e.Action == NotifyCollectionChangedAction.Remove || e.Action == NotifyCollectionChangedAction.Reset)
            {
                foreach (var item in e.OldItems)

```

```
        [Test]
        public async Task TestAddingARecordToList()
        {
            var name = Guid.NewGuid().ToString();
            var key = "test_record";
            var listKey = "test_list";
            using (var watchClient = await GetConnection("chris", "test"))
            {
                var watchList = await watchClient.Records.GetListAsync(listKey);
                var changes = new List<NotifyCollectionChangedEventArgs>();
                watchList.CollectionChanged += (sender, args) =>
                {
                    changes.Add(args);
                };

                using (var updateClient = await GetConnection("admin", "test"))
                {
                    var list = await updateClient.Records.GetListAsync(listKey);

                    var rec = await updateClient.Records.GetRecordAsync(key);
                    var res = await updateClient.Records.SetWithAckAsync(rec, "name", name);
                    Assert.IsTrue(res);
                    list.Add(rec.RecordName);
                }
                while (changes.Count == 0)
                {
                    await Task.Delay(300);
                }

            }
        }

```

**Comment by schulz3000** [comment]
this should be already fixed with 4deaf4745afbeb5fbb02a104883d76eac2fd9382

**Comment by grendo** [comment]
with the patch from https://github.com/schulz3000/deepstreamNet/issues/29 and 4deaf47 it now works
